SeafFileProvider/SeafEnumerator.m

```diff
@@ -82,8 +84,15 @@ - (void)enumerateItemsForObserver:(id<NSFileProviderEnumerationObserver>)observe
             
             SeafDir *dir = (SeafDir *)[_item toSeafObj];
             [dir loadContentSuccess: ^(SeafDir *d) {
-                [observer didEnumerateItems:[self getSeafDirProviderItems:d startingAtPage:page]];
-                [observer finishEnumeratingUpToPage:nil];
+                NSArray *items = [self getSeafDirProviderItems:d startingAtPage:page];
+                [observer didEnumerateItems:items];
+                if (items.count == self.maxItemCount) {
+                    NSInteger numPage = [[NSString stringWithUTF8String:[page bytes]] integerValue] + 1;
+                    NSData *providerPage = [[NSString stringWithFormat:@"%ld", (long)numPage] dataUsingEncoding:NSUTF8StringEncoding];
+                    [observer finishEnumeratingUpToPage:providerPage];
+                } else {
```

Comment:
Use items.count != self.maxItemCount to verify if it reaches the end might not work well. What if the last chunk has exactly self.maxItemCount items.